src/datasources/accounts/accounts.datasource.ts

```diff
@@ -66,27 +69,34 @@ export class AccountsDatasource implements IAccountsDatasource, OnModuleInit {
     );
   }
 
+  // TODO: document this function explaining why the name is hashed (unique constraint)
   async createAccount(args: {
-    address: `0x${string}`;
+    createAccountDto: CreateAccountDto;
     clientIp: string;
   }): Promise<Account> {
     await this.checkCreationRateLimit(args.clientIp);
+    const { address, name } = args.createAccountDto;
+    const hash = crypto.createHash('sha256');
+    hash.update(name);
+    const nameHash = hash.digest('hex');
+    // TODO: encrypt the name
```

Comment:
The actual name encryption will be added in a forthcoming PR, along with AWS KMS support, which is needed to encrypt the name.